### PR TITLE
Feature/.collect() method

### LINKED ISSUE #21: Implement `collect` method
https://ruby-doc.org/core-2.6/Array.html#method-i-collect

### PR DESCRIPTION
**Pull request by arnellebalane**
Closes #21

### CODE REVIEW

File: methods/collect/tests.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    t.deepEqual(
+        collect([1, 2, 3], x => x * 2),
+        [2, 4, 6]
+    );
+
+    t.deepEqual(
+        collect(['a', 'b', 'c'], x => `${x}!`),
+        ['a!', 'b!', 'c!']
+    );
+});
```
Comment (aldnav) [inline methods/collect/tests.js]: @arnellebalane In Ruby, when no block is given (or no callback is provided), I get this.

```ruby
irb(main):001:0> a = ["a", "b", "c"]
=> ["a", "b", "c"]
irb(main):002:0> a.collect
=> #<Enumerator: ["a", "b", "c"]:collect>
```

And in rbjs,  
```js

collect([1, 2, 3])

  1 test failed

  return values returned by the given function when called with each array item

  /home/aldnav/Pro/rbjs/methods/collect/index.js:22

   21: export default function collect(array, callback) {
   22:     return array.map(callback);
   23: }

  Error thrown in test:

  TypeError {
    message: 'undefined is not a function',
  }
```

What is the expected behavior when callback is undefined? Do we return a copy of the array?

```js
t.deepEqual(
    collect([1, 2, 3]),
    [1, 2, 3]  // or is this 'null'?
);
```

Comment (arnellebalane) [reply methods/collect/tests.js]: @aldnav Oh right my bad, my implementation only assumes that a callback is provided. Will have to modify that one. Thanks for spotting that one 😄 

We'll definitely want to support a missing callback, not sure how we'd like to go with it? I wonder what the equivalent of a Ruby `Enumerator` in JS would be. Will do some more research.

Comment (aldnav) [reply methods/collect/tests.js]: @arnellebalane on top of my head is to return the same exact array but I'm not sure if it's right.

Comment (arnellebalane) [reply methods/collect/tests.js]: @aldnav Looks like Ruby's `Enumerator` is similar in nature to JavaScript's iterators, they both allow iteration of the object's items. A JS array is already iterable, so returning the input array as per your suggestion would make sense 👍

Comment (arnellebalane) [reply methods/collect/tests.js]: @aldnav PR has been updated with this functionality.

Comment (aldnav) [reply methods/collect/tests.js]: @arnellebalane update looks good! 👍

**Pull request merged**